### DISCUSSION TITLE
Is it possible to query a local working copy of the ontology?

### DISCUSSION THREAD

**Discussion by bbartley** [Bioprotocols/container-ontology#19] [Q&A] [answered]
I haven't totally deciphered the Makefile, but I'm guessing queries are directed at a web server somewhere?  In any event, my local changes to the ontology file aren't returned by queries.  Is there a way to query the local turtle file?

**Comment by rpgoldman** [comment, marked as answer]
No, I'm afraid not, because we want information from the closure of the Turtle file that is computed by the server.

It would be possible to precompute and store it, but that's not done now.

Doing `make server` should give you a running server and make the queries go through.

I don't have any great ideas about how to ensure that the container server is started up when needed; if you do, please let me know. (e.g., the container server might be remote, so we can't just always start it up when we load the container API)

**Reply by bbartley** [reply]
Let me ask a clarifying question:  When I run `make server`, does it import the ontology file from my local project directory, or does it fetch the ontology from a remote server somewhere?

**Reply by rpgoldman** [reply]
It pulls it from your local drive.

I still have an IOU to myself to augment the setup so that you can point the container server at your own list of ontology files.

This is doable RN (ask me how if you are interested -- but please do it in a separate question so others can easily find the answer), but not well supported.
